custom_components/pyscript/trigger.py

```diff
@@ -739,7 +739,17 @@ async def trigger_watch(self):
                 elif notify_type == "state":
                     new_vars, func_args = notify_info
 
-                    if "var_name" not in func_args or func_args["var_name"] not in self.state_trig_ident_any:
+                    # check if any state_trig_ident_any starts with var_name
+                    any_match = False
+                    if "var_name" in func_args:
+                        for check_var in self.state_trig_ident_any:
+                            if check_var == func_args['var_name']:
+                                any_match = True
+                                break
+                            if check_var.startswith(f"{func_args['var_name']}."):
+                                any_match = True
+                                break
+                    if not any_match:
```

Comment:
Somehow my longest and most important comment got lost, so repeating it here.  Basically all the checks below should move up above this line, so that the trigger function is only evaluated if necessary.  Specifically, it should be skipped altogether if any of the three "any" cases are true:
- state value change when `state_trig_ident_any` contains `var_name`
- attribute change for any entry in `state_trig_ident_any` of the form `var_name.attr`
- any attribute change if an entry in `state_trig_ident_any` is of the form `var_name.*`

Next, you then should move the code that checks `self.state_trig_ident` to see if any of mentioned variables have changed.  Only if that is the case should the trigger function be evaluated.